### PR TITLE
Fixed Issue #2181

### LINKED ISSUE #2181: pet_db2 overrides pet_db
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Issue Prelude

[//]: # (Please complete these mandatory steps and check the following boxes by putting an `x` inside the brackets _before_ filing your issue)

-  [x] I have not modified the source prior to reproducing this issue.
-  [x] I am using the latest version of Hercules.
-  [x] I am aware that this report will be closed or deleted if it becomes obvious that I am stating the false.

### Description
Whenever I define anything in `pet_db2.conf`, it causes everything in `pet_db.conf` to break.

### Expected Behavior
pet_db2 should be additions to pet_db, or have some sort of inherit flag like item_db2 does.

**Branch(es):**
- [x] master
- [x] stable

**Hercules rev. hash/commit:** 

Git revision src:

[//]: # (Copy the first 3 lines of the login-server, char-server or map-server startup.)
[//]: # ( [Info]: Hercules 64-bit for Mac OS X )
[//]: # ( [Info]: Git revision src: 'a5918b329ca0826b04dca32ede783586403f58db' )
[//]: # ( [Info]: Git revision scripts: 'a5918b329ca0826b04dca32ede783586403f58db' )

### Operating System

[//]: # (Mac OS X 10.12.3 16D32 [x86_64])
[//]: # (Thank you for adhering to this process! It ensures your issue is resolved quickly and that neither your nor our time is needlessly wasted.)
[//]: # (This template is for problem reports. For other types of report, edit it accordingly.)
[//]: # (For fixes please create a Pull Request.)

### PR DESCRIPTION
**Pull request by dastgirp**
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Pull Request Prelude

[//]: # (Thank you for working on improving Hercules!)

[//]: # (Please complete these steps and check the following boxes by putting an `x` inside the brackets _before_ filing your Pull Request.)

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

[//]: # (Describe at length, the changes that this pull request makes.)

PetDB2 no longer overrides entry of PetDB

**Affected Branches:** 

[//]: # (Master? Slave?)

**Issues addressed:**

[//]: # (Issue Tracker Number if any.)
Closes #2181 

### Known Issues and TODO List

[//]: # (Insert checklist here)
[//]: # (Syntax: - [ ] Checkbox)

[//]: # (**NOTE** Enable the setting "[√] Allow edits from maintainers." when creating your pull request if you have not already enabled it.)

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2206)

File: src/map/pet.c [hunk 1]
```diff
@@ -1254,23 +1254,23 @@ static void pet_read_db(void)
 		DBPATH"pet_db.conf",
 		"pet_db2.conf"
 	};
-	int i;
+	int i, count = 0;
```
Comment (4144) [inline src/map/pet.c]: while you changing this line, ``int i``, better put inside for as ``for (int i = 0 ...``, before it was impossible due outdated visual studio version.

**Comment by Asheraf** [comment]
Do we want this not to override? it looks inconsistent with what we have with other databases (item, mob).

**Comment by EyesOfAHawk** [comment]
@Asheraf it makes pet_db.conf not exist at the moment. So if you put 1 entry in pet_deb2, even if its unrelated, now all the pet_db entries dont work.

**Comment by Asheraf** [comment]
@EyesOfAHawk that is because it's using the entry index in the file as the array index unlike other databases which use the id as key, currently with this pr if you add a duplicated pet id into the 2nd file it will just get ignored.

**Comment by MishimaHaruna** [comment]
This should probably go in after #2600 since it would make it easier to address the override issue (the unique id to use for overriding could be the egg id, since it becomes a mandatory field, and it doesn't make any sense to have multiple pets with the same egg)

**Comment by Kenpachi2k13** [comment]
> and it doesn't make any sense to have multiple pets with the same egg)

It's not really related to this PR, but reminds me of these two pets which share the same egg:
```
{
	Id: 2057
	Name: "Strange Cramp"
	EggItem: "Mystic_Leaf_Cat_Ball"
	TamingItem: "Leaf_Cat_Ball"
	FoodEffectiveness: 1
	HungerDecrement: 0
	Intimacy: {
		FeedIncrement: 1
		OverFeedDecrement: 0
		OwnerDeathDecrement: 0
	}
	CaptureRate: 5000
	AttackRate: 350
	DefendRate: 400
},
{
	Id: 2081
	Name: "Strange Hydra"
	EggItem: "Mystic_Leaf_Cat_Ball"
	TamingItem: "Leaf_Cat_Ball"
	FoodEffectiveness: 1
	HungerDecrement: 0
	Intimacy: {
		FeedIncrement: 1
		OverFeedDecrement: 0
		OwnerDeathDecrement: 0
	}
	CaptureRate: 5000
	AttackRate: 350
	DefendRate: 400
},
```

**Comment by MishimaHaruna** [comment]
Good point @Kenpachi2k13 , I was not aware of that. Then, we'll just use `Id`. I'll see if I manage to check this for the next release, to make it behave in a similar way to how the item/mob databases can override/inherit. Otherwise, anyone else can grab this if they feel like it

**Comment by MishimaHaruna** [comment]
I rebased this and implemented an inheritance mechanism similar to the mob db one (based on the pet mob id, which is an unique key for the pet db)

**Comment by 4144** [comment]
look like hpm not updated

**Pull request merged**